tests/unit/acquisition/multi_objective/test_function.py

```diff
@@ -617,3 +622,119 @@ def prepare_acquisition_function(
+def test_hippo_penalizer_raises_for_invalid_batch_size(at: TensorType) -> None:
+    pending_points = tf.zeros([1, 2], dtype=tf.float64)
+    hp = hippo_penalizer(QuadraticMeanAndRBFKernel(), pending_points)
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        hp(at)
+
+
+def test_hippo_penalizer_raises_for_empty_pending_points() -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        hippo_penalizer(QuadraticMeanAndRBFKernel(), None)
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        hippo_penalizer(QuadraticMeanAndRBFKernel(), tf.zeros([0, 2]))
+
+
+def test_hippo_penalizer_update_raises_for_empty_pending_points() -> None:
+    pending_points = tf.zeros([1, 2], dtype=tf.float64)
+    hp = hippo_penalizer(QuadraticMeanAndRBFKernel(), pending_points)
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        hp.update(None)
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        hp.update(tf.zeros([0, 2]))
+
+
+@pytest.mark.parametrize(
+    "point_to_penalize", [tf.constant([[[0.0, 1.0]]]), tf.constant([[[3.0, 4.0]]])]
+)
+def test_hippo_penalizer_penalizes_pending_point(point_to_penalize: TensorType) -> None:
+    pending_points = tf.constant([[0.0, 1.0], [2.0, 3.0], [3.0, 4.0]])
+    hp = hippo_penalizer(QuadraticMeanAndRBFKernel(), pending_points)
+
+    penalty = hp(point_to_penalize)
+
+    # if the point is already collected, it shall be penalized to 0
+    npt.assert_allclose(penalty, tf.zeros((1, 1)))
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "base_builder",
+    [
+        ExpectedHypervolumeImprovement().using(""),
+        ExpectedConstrainedHypervolumeImprovement("", _Certainty(), 0.0),
+    ],
+)
+def test_hippo_penalized_acquisitions_match_base_acquisition(
+    base_builder: AcquisitionFunctionBuilder[ProbabilisticModel],
+) -> None:
+    data = {"": Dataset(tf.zeros([3, 2], dtype=tf.float64), tf.ones([3, 2], dtype=tf.float64))}
+    model = {"": _mo_test_model(2, *[None] * 2)}
+
+    hippo_acq_builder = HippoPenalizationAcquisitionFunction(
+        "", base_acquisition_function_builder=base_builder
+    )
+    hippo_acq = hippo_acq_builder.prepare_acquisition_function(model, data, None)
+
+    base_acq = base_builder.prepare_acquisition_function(model, data)
+
+    x_range = tf.linspace(0.0, 1.0, 11)
+    x_range = tf.cast(x_range, dtype=tf.float64)
+    xs = tf.reshape(tf.stack(tf.meshgrid(x_range, x_range, indexing="ij"), axis=-1), (-1, 2))
+    hippo_acq_values = hippo_acq(xs[..., None, :])
+    base_acq_values = base_acq(xs[..., None, :])
+
+    npt.assert_array_equal(hippo_acq_values, base_acq_values)
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "base_builder",
+    [
+        ExpectedHypervolumeImprovement().using(""),
+        ExpectedConstrainedHypervolumeImprovement("", _Certainty(), 0.0),
+    ],
+)
+def test_hippo_penalized_acquisitions_combine_base_and_penalization_correctly(
+    base_builder: AcquisitionFunctionBuilder[ProbabilisticModel],
```

Comment:
Could you add a test to check retracing? We have some for other acquisition functions

Reply:
will look into adding one for it

Reply:
added